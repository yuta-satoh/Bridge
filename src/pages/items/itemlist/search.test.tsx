import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Search from "./search";

jest.mock('next/router', () => ({
    useRouter() {
      return {
        replace: jest.fn(),
        push: jest.fn(),
        reload: jest.fn(),
      };
    },
}));

const filter = [
    {
        id: 1,
        name: "ベルベットアームチェア",
        description: "ベルベットのお洒落なデザインチェア。",
        genre: "フェミニン",
        category: "椅子",
        price: 16000,
        imgurl: "/images/chair/chair_feminine_1.jpg",
        stock: 72,
        delete: false
    },
    {
        id: 2,
        name: "丸椅子スツール",
        description: "落ち着いた可愛らしいベロア風スツール。",
        genre: "フェミニン",
        category: "椅子",
        price: 10000,
        imgurl: "/images/chair/chair_feminine_2.jpg",
        stock: 99,
        delete: false
    },
]

const maxPage = '1'

const nowOrder = {
    genres: '北欧',
    categories: 'ソファ',
    input: 'スツール',
    order: 'new',
    page: '0',
}


describe('検索ページ', () => {
    test('正常に描画される', () => {
        const { asFragment } = render(<Search filter={filter} maxPage={maxPage} nowOrder={nowOrder} />);
        expect(asFragment()).toMatchSnapshot()
    })

    test('検索結果が無い', () => {
        const { asFragment } = render(<Search filter={[]} maxPage={maxPage} nowOrder={nowOrder} />);
        expect(asFragment()).toMatchSnapshot()
    })
    
    test('並び替えのセレクトボックスが選択可能か', async () => {
        const user = userEvent.setup();
        render(<Search filter={filter} maxPage={maxPage} nowOrder={nowOrder} />);
        const selectOrder = screen.getByTestId('select-option');
        await user.selectOptions(selectOrder, '高い順')
        const expensiveSelect = screen.getByRole('option', { name: '高い順' }) as HTMLOptionElement
        expect(expensiveSelect.selected).toBeTruthy()
    })
})
