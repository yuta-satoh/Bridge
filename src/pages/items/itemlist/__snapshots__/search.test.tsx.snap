// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`検索ページ 検索結果が無い 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <div
      class="Breadcrumb"
    >
      <nav>
        <ol>
          <li
            class="breadlist"
          >
            <a
              href="/"
            >
              TOPページ
            </a>
          </li>
          <li
            class="breadlist"
          >
            <a
              href="/items/itemlist"
            >
              商品一覧ページ
            </a>
          </li>
          <li
            class="breadlist"
          >
            検索結果
          </li>
        </ol>
      </nav>
    </div>
    <div
      class="result_none"
    >
      <p>
        検索結果がありません
      </p>
      <a
        href="/items/itemlist"
      >
        商品一覧に戻る
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`検索ページ 正常に描画される 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <div
      class="Breadcrumb"
    >
      <nav>
        <ol>
          <li
            class="breadlist"
          >
            <a
              href="/"
            >
              TOPページ
            </a>
          </li>
          <li
            class="breadlist"
          >
            <a
              href="/items/itemlist"
            >
              商品一覧ページ
            </a>
          </li>
          <li
            class="breadlist"
          >
            検索結果
          </li>
        </ol>
      </nav>
    </div>
    <div
      class="text-right w-full"
    >
      <label
        for="itemOrder"
      >
        表示順：
      </label>
      <label
        class="selectbox"
      >
        <select
          data-testid="select-option"
          id="itemOrder"
          name="itemOrder"
        >
          <option
            value="新着順"
          >
            新着順
          </option>
          <option
            value="安い順"
          >
            安い順
          </option>
          <option
            value="高い順"
          >
            高い順
          </option>
        </select>
      </label>
    </div>
    <div
      class="list_outer"
    >
      <div>
        <a
          href="/items/itemlist/1"
        >
          <div
            class="image"
          >
            <img
              alt="ベルベットアームチェア"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=3840&q=75"
              srcset="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_1.jpg&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="detail"
          >
            <p
              class="itemname"
            >
              ベルベットアームチェア
            </p>
            <p>
              ¥ 17,600
            </p>
            <p
              class="itemdescription"
            >
              ベルベットのお洒落なデザインチェア。
            </p>
          </div>
        </a>
      </div>
      <div>
        <a
          href="/items/itemlist/2"
        >
          <div
            class="image"
          >
            <img
              alt="丸椅子スツール"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_2.jpg&w=3840&q=75"
              srcset="/_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_2.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_2.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_2.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_2.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_2.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_2.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_2.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fchair%2Fchair_feminine_2.jpg&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
          <div
            class="detail"
          >
            <p
              class="itemname"
            >
              丸椅子スツール
            </p>
            <p>
              ¥ 11,000
            </p>
            <p
              class="itemdescription"
            >
              落ち着いた可愛らしいベロア風スツール。
            </p>
          </div>
        </a>
      </div>
    </div>
    <div
      class="paging"
    >
      <ul
        class="pages"
      >
        <li
          class="currentPage"
        >
          <button
            class="button"
            type="button"
          >
            1
          </button>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
